### PR TITLE
first siam_rpn_test

### PR DESCRIPTION
**Pull request by FrankYoungchen**
* test.py in scripts/tracking
        * benchmark.py in scripts/tracking
        * dataloader in gluoncv/data/otb
        * nerwork with Simplified alexnet in gluoncv/model_zoo/siamrpn/siamrpn_tracker.py
        * generate_anchor or IOU in gluoncv

### CODE REVIEW

File: gluoncv/data/otb/tracking.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        self.attr = {}
+        self.attr['ALL'] = list(self.videos.keys())
+        for x in attr:
+            self.attr[x] = []
+        for k, v in self.videos.items():
+            for attr_ in v.attr:
+                self.attr[attr_].append(k)
+
+if __name__ == '__main__':
+    OTBDataset(name='OTB2015',dataset_root='/home/ubuntu/cyk/tracking/tracking_test_2/testing_dataset/OTB2015/',load_img=False)
```
Comment (bryanyzhu) [inline gluoncv/data/otb/tracking.py]: It is better to remove specific path on your machine, such as `/home/ubuntu/cyk/tracking/tracking_test_2/testing_dataset/OTB2015/`. Usually we ignore it or we use a default location at `~/.mxnet/datasets/`.

File: gluoncv/model_zoo/siamrpn/siam_alexnet.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# coding: utf-8
+from __future__ import division
+from mxnet.gluon import nn
+from mxnet.gluon.block import HybridBlock
+
+class AlexNetLegacy(HybridBlock):
+    configs = [3, 96, 256, 384, 384, 256]
+
+    def __init__(self,width_mult=1,**kwargs):
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/siamrpn/siam_alexnet.py]: Add white spaces in between, use pylint to check the code style first.

File: gluoncv/model_zoo/siamrpn/siam_rpn.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                nn.BatchNorm(),
+                nn.Activation('relu'),
+                nn.Conv2D(out_channels, kernel_size=1)
+                )
+        
+
+    def hybrid_forward(self,F, kernel, search):
+        kernel = self.conv_kernel(kernel)#[1, 256, 24, 24]
+        search = self.conv_search(search)#[1, 256, 4, 4]
+        batch = kernel.shape[0]
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/siamrpn/siam_rpn.py]: In hybrid mode, we can't use `.shape` get shape anymore. We can discuss offline how to bypass this issue.

File: gluoncv/model_zoo/siamrpn/siam_rpn.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    def forward(self, z_f, x_f):
+        cls = self.cls(z_f, x_f)
+        loc = self.loc(z_f, x_f)
+        cls = self.cls(z_f, x_f)
+        return cls, loc
+
+if __name__ == '__main__':
+    model = DepthwiseRPN()
+    print(model)
+    # model.initialize()
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/siamrpn/siam_rpn.py]: It is better to remove comments after you finish debugging.

File: gluoncv/utils/metrics/tracking.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
+import numpy as np
+from colorama import Style, Fore
+
+def overlap_ratio(rect1, rect2):
```
Comment (bryanyzhu) [inline gluoncv/utils/metrics/tracking.py]: For each metric, please add more description, such as what is the metric, which paper proposes this metric, what is the input, what is the data type, what is the default value, what is the output, etc. Because usually these metrics are shared by many tasks or datasets. Keep a good documentation will help others contribute to GluonCV.

File: scripts/tracking/benchmark.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+parser.add_argument('--tracker_path', '-p',type=str,help='tracker result path')
+parser.add_argument('--dataset', '-d', default='OTB2015',type=str,help='dataset name')
+parser.add_argument('--num', '-n', default=1, type=int,help='number of thread to eval')
+parser.add_argument('--tracker_prefix', '-t',type=str, help='tracker name')
+parser.add_argument('--show_video_level', '-s', dest='show_video_level',action='store_true')
+parser.add_argument('--test_dataset',type=str,help='test_json dataset dir')
+parser.set_defaults(show_video_level=False)
+args = parser.parse_args()
+
+def benchmark():
```
Comment (bryanyzhu) [inline scripts/tracking/benchmark.py]: What is the difference between `test.py` and `benchmark.py`? Is there a good reason we separate them?

Comment (FrankYoungchen) [reply scripts/tracking/benchmark.py]: test.py output is *.txt, where one video corresponds to one txt,which save in 'result'.
benchmark.py is recording evaluation such as success or Precision, which according to output and label.
I think it's convenient to add SiamMask and Pyost does this，and it's OK if merge them.

Comment (FrankYoungchen) [reply scripts/tracking/benchmark.py]: **test.py** output is *.txt, where one video corresponds to one txt,which save in 'result'.
benchmark.py is recording evaluation such as success or Precision, which according to output and label.
I think it's convenient to add SiamMask and Pyost does this，and it's OK if merge them.

Comment (FrankYoungchen) [reply scripts/tracking/benchmark.py]: ### one

Comment (FrankYoungchen) [reply scripts/tracking/benchmark.py]: _`one`_

Comment (FrankYoungchen) [reply scripts/tracking/benchmark.py]: _test.py_ output is some *.txt, where one video corresponds to one txt, which records some output.
_benchmark.py_  records some Evaluation  such as Success or Precision according to output txt and label.
I think it's convenient to add SiamRPN++ or SiamMask and pysot does it，and it's OK if merge them

Comment (FrankYoungchen) [reply scripts/tracking/benchmark.py]: **test.py** output is some *.txt, where one video corresponds to one txt, which records some output.
**benchmark.py** records some Evaluation such as Success or Precision according to output txt and label.
I think it's convenient to add SiamRPN++ or SiamMask and pysot does it，and it's OK if merge them

Comment (FrankYoungchen) [reply scripts/tracking/benchmark.py]: ### test.py output is some *.txt, where one video corresponds to one txt, which records some output.
### benchmark.py records some Evaluation such as Success or Precision according to output txt and label.
I think it's convenient to add SiamRPN++ or SiamMask and pysot does it，and it's OK if merge them

Comment (FrankYoungchen) [reply scripts/tracking/benchmark.py]: `test.py` output is some *.txt, where one video corresponds to one txt, which records some output.
`benchmark.py` records some Evaluation such as Success or Precision according to output txt and label.
I think it's convenient to add SiamRPN++ or SiamMask and pysot does it，and it's OK if merge them

Comment (bryanyzhu) [reply scripts/tracking/benchmark.py]: Ok, I understand. I think it's good to have both files, let's keep it as it is.

**Comment by bryanyzhu** [comment]
This PR was re-opened in #1139 , will close this one.
